src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -0,0 +1,110 @@
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateTestExecutionScripts : Task
+    {
+        [Required]
+        public string[] TestCommands { get; set; }
+
+        [Required]
+        public string[] TestDependencies { get; set; }
+
+        [Required]
+        public string RunnerScriptTemplate { get; set; }
+
+        [Required]
+        public string ScriptOutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(RunnerScriptTemplate))
+            {
+                throw new FileNotFoundException($"Runner script template {RunnerScriptTemplate} was not found.");
+            }
+
+            string executionScriptTemplate = File.ReadAllText(RunnerScriptTemplate);
+            Directory.CreateDirectory(Path.GetDirectoryName(ScriptOutputPath));
+
+            Log.LogMessage("Test Command lines = {0}", string.Join("\n", TestCommands));
+            string extension = Path.GetExtension(Path.GetFileName(ScriptOutputPath)).ToLowerInvariant();
+            switch (extension)
+            {
+                case ".sh":
+                    WriteShExecutionScript(executionScriptTemplate, ScriptOutputPath);
+                    break;
+                case ".cmd":
+                case ".bat":
+                    WriteCmdExecutionScript(executionScriptTemplate, ScriptOutputPath);
+                    break;
+                default:
+                    throw new System.NotSupportedException($"Generating runner scripts with extension '{extension}' is not yet supported");
+            }
+            return true;
+        }
+
+        private void WriteShExecutionScript(string shExecutionTemplate, string outputPath)
+        {
+            // Build up the copy commands... 
+            StringBuilder copyCommands = new StringBuilder();
+            foreach (string dependency in TestDependencies)
+            {
+                string normalizedDependency = dependency.Replace('\\', '/');
+                if (normalizedDependency.StartsWith("/"))
+                {
+                    normalizedDependency = normalizedDependency.Substring(1);
+                }
+                copyCommands.Append($"cp -l -f \"$PACKAGE_DIR/{normalizedDependency}\" \"$EXECUTION_DIR/{Path.GetFileName(dependency)}\" || exit $?\n");
+            }
+            shExecutionTemplate = shExecutionTemplate.Replace("[[CopyFilesCommands]]", copyCommands.ToString());
+
+            StringBuilder testRunCommands = new StringBuilder();
+            foreach (string runCommand in TestCommands)
+            {
+                // Debugging: Trying to figure out why this happpens currently but working around it.
+                if (runCommand.Contains("CoreRun.exe"))
+                {
+                    Log.LogMessage("For debug: Adding Run Command: {0}", runCommand);
+                }
+                
+                testRunCommands.Append(runCommand.Replace("CoreRun.exe", "./corerun"));
```

Comment:
I wouldn't expect the test commands to have need this treatment. Why wouldn't we be getting the correct exe name already?